pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -689,6 +701,26 @@ func (vm *VirtualMachine) GetGpuPartitionSettingData(partitionSizeBytes uint64)
 	return
 }
 
+func (vm *VirtualMachine) GetDefaultGpuPartitionSettingData() (partitionSettingData *gpu.GpuPartitionSettingData, err error) {
+	settings, err := vm.GetVirtualSystemSettingData()
+	if err != nil {
+		return
+	}
+	defer settings.Close()
+
+	gpuPartitionSettingCollection, err := vm.GetGpuPartitionSettingCollection()
+	if err != nil {
+		return nil, err
+	}
+	defer gpuPartitionSettingCollection.Close()
+
+	if len(gpuPartitionSettingCollection) == 0 {
+		err = errors.Wrapf(errors.NotFound, "Unable to find GPU partition assigned to vm [%+v]", vm)
+	}
+
+	return gpuPartitionSettingCollection[0], nil
```

Comment:
This pattern is not correct.
Since we are closing the collection, we cannot return this object.
Create a clone of new partitionsettingdata and return that clone

Reply:
Fixed.